### PR TITLE
unreal_asset api upstream merge

### PR DESCRIPTION
**Pull request by localcc**
UAssetAPI commit: `48258cbeceda62f6fc24622df103d2f07e6cc77f`

### CODE REVIEW

File: unreal_asset/src/kismet/mod.rs [hunk 1]
```diff
@@ -3,16 +3,17 @@ use std::fmt::Debug;
 use std::hash::Hash;
 use std::mem::size_of;
 
-use byteorder::LittleEndian;
+use byteorder::{LittleEndian, LE};
```
Comment (konsti219) [inline unreal_asset/src/kismet/mod.rs:6]: decide for one

File: unreal_asset/src/lib.rs [hunk 2]
```diff
@@ -1568,185 +711,62 @@ impl<'a, C: Read + Seek> Asset<C> {
...
-            // todo: warning?
-
-            self.cursor
-                .seek(SeekFrom::Start(base_export.serial_offset as u64))?;
-            return Ok(RawExport::from_base(base_export.clone(), self)?.into());
-        } else if let Some(normal_export) = export.get_normal_export_mut() {
-            let mut extras = vec![0u8; extras_len as usize];
-            self.cursor.read_exact(&mut extras)?;
-            normal_export.extras = extras;
+            self.asset_data.exports = new_exports;
```
Comment (konsti219) [inline unreal_asset/src/lib.rs:762]: Use `Vec::retain` to avoid reallocating.

**Review by konsti219 — CHANGES_REQUESTED** [review]
image
:joedead:

File: unreal_asset/src/lib.rs (hunk 2 above)
Comment (localcc) [reply unreal_asset/src/lib.rs:762]: I'm not really sure how `Vec::retain` helps here, as all the exports will be replaced with the new ones, there will be nothing to retain

Comment (localcc) [reply unreal_asset/src/lib.rs:762]: Doing something like map would require somehow getting results out to throw an error if something went wrong

**Review by konsti219 — APPROVED** [review]

**Pull request merged**